Core/clim-basic/drawing/medium.lisp

```diff
@@ -216,10 +218,15 @@
                                        (text-style text-style)
                                        &optional character-set)
   (declare (ignore character-set))
-  ;; Cache mappings.
-  (ensure-gethash (parse-text-style text-style)
-                  (port-text-style-mappings port)
-                  (call-next-method)))
+  ;; Cache mappings only for fully specified text styles.
+  (if (or (device-font-text-style-p text-style)
+          (multiple-value-bind (family face size)
+              (clim:text-style-components text-style)
+            (and family face size)))
+      (ensure-gethash text-style
+                      (port-text-style-mappings port)
+                      (call-next-method))
+      (call-next-method)))
```

Comment:
Could we move the `parse-text-style*` call that many (all?) backends do into this else branch?

Reply:
parse-text-style* is just an utility defined for ttf renderer, backends are free to use it but I don't think it belongs here.